[Shipping Labels] Update buttons logic on hazmat selection screen

irfano opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->
Closes WOOMOB-580

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR adds a save button and updates the visibility logic of the "Select Category" button on the Hazmat selection screen.

- The Save button becomes visible after unchecking the checkbox. If “unchecked” is already the saved state, the Save button will not be shown.
- The Select category button is hidden when a hazmat option is already selected.

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
1. Go to Orders.
2. Select an order that is eligible to create shipping labels.
3. Tap Create shipping labels button.
4. Tap Hazmat selection.
<img width="292" height="73" alt="image" src="https://github.com/user-attachments/assets/b9478b07-7263-4661-a7d2-41652171a577" />

5. Check "Contains hazardous materials" checkbox.
6. Tap "Select a category" button and select a category from the next screen.
7. Tap Hazmat selection again.
8. Uncheck the selection. Verify Save button is visible and "Select Category" button is hidden.
9. Check the checkbox. Verify Save button is hidden and "Select Category" button is visible.
10. Tap Save.
11. Tap Hazmat selection again.
12. Uncheck the selection.
13. Tap back button and verify the hazmat option is still selected.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
Steps above.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
[Screen_recording_20250820_123238.webm](https://github.com/user-attachments/assets/4e2f7ecd-32c8-42dc-8407-a530352be475)

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

irfano committed:
Update Material imports in WooShippingLabelHazmatFormScreen

The `Icon` and `IconButton` imports are changed from `androidx.compose.material` to `androidx.compose.material3`.

irfano committed:
Add "Save" button to Hazmat form

This commit introduces a "Save" button to the `WooShippingLabelHazmatFormScreen`.

irfano committed:
Remove the logic for back press in hazmat form view model

irfano committed:
Add hazmat category selection saving

This commit introduces the ability to save the selected hazmat category.

irfano committed:
Manage save button visibility in WooShippingLabelHazmatFormViewModel

irfano committed:
Update WooShippingLabelHazmatFormViewModelTest

irfano committed:
Add hazmat select category button visibility logic

This commit introduces a new property `isSelectCategoryButtonVisible` to the `ViewState` data class in `WooShippingLabelHazmatFormViewModel.kt`.

irfano committed:
Use isSelectCategoryButtonVisible logic in hazmat selection screen

irfano committed:
Add tests for hazmat category button visibility

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/265">23.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

irfano committed:
Update RELEASE-NOTES.txt

The release notes are updated to include a change to the button logic on the Hazardous Materials screen for Shipping Labels.

irfano committed:
Merge branch 'trunk' into issue/WOOMOB-580-add-save-button-to-hazmat-screen

hichamboushaba reviewed (approved):
LGTM, thanks for addressing this @irfano

irfano committed:
Merge branch 'trunk' into issue/WOOMOB-580-add-save-button-to-hazmat-screen

irfano committed:
Merge branch 'trunk' into issue/WOOMOB-580-add-save-button-to-hazmat-screen
